core/src/impl/Kokkos_Default_GraphNodeKernel.hpp

```diff
@@ -36,15 +36,22 @@ struct GraphNodeKernelDefaultImpl {
   // TODO @graphs decide if this should use vtable or intrusive erasure via
   //      function pointers like in the rest of the graph interface
   virtual void execute_kernel() = 0;
+
+  GraphNodeKernelDefaultImpl() = default;
+
+  explicit GraphNodeKernelDefaultImpl(ExecutionSpace exec)
+      : m_execution_space(std::move(exec)) {}
```

Comment:
Any reason not to take the execution space instance by const reference as we usually do?

Reply:
Noting that:
1.  `Kokkos` execution space classes are behaving as shared pointers to an internal resource,
2.  `GraphNodeKernelDefaultImpl` needs to explicitly extend the lifetime of the underlying internal execution space instance,

I'm using http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r34-take-a-shared_ptrwidget-parameter-to-express-shared-ownership.

Clearly not the standard way of doing things in `Kokkos` for sure, but definitely worth a thought.

Reply:
> Kokkos execution space classes are behaving as shared pointers to an internal resource

That's an internal detail. I'm not sure if that's what we want to tell people and teach them to do.
I won't block on this question.